gpu: Update setup documentation (#33090)
DataDog/documentation: content/en/gpu_monitoring/setup.md

title: Set up GPU Monitoring
private: true
---
This page provides instructions on setting up Datadog's GPU Monitoring on your infrastructure. The configuration instructions depend on the environment to monitor.

### Prerequisites

To begin using Datadog's GPU Monitoring, your environment must meet the following criteria:

- You are a Datadog user with active Datadog infrastructure hosts. If not, see the following guides:
  - [Install the Datadog Agent on Kubernetes][1]
  - [Install the Datadog Agent on Docker][7]
  - [Install the Datadog Agent on non-containerized Linux][8]
- The NVIDIA drivers are installed on the hosts. If using Kubernetes, the NVIDIA device plugin for Kubernetes is installed ([directly][3], or through [NVIDIA GPU Operator][4])

#### Minimum version requirements

> - **Datadog Agent**: version 7.72.2

Reviewer: ```suggestion
- **Datadog Agent**: v7.72.2
```

- **Operating system**: Linux
   - (Optional) For advanced eBPF metrics, Linux kernel version 5.8
- **NVIDIA driver**: version 450.51

If using Kubernetes, the following requirements must be met:

- [**Datadog Operator**][5]: version 1.18, _or_ [**Datadog Helm chart**][6]: version 3.137.3
- **Kubernetes**: 1.22 with PodResources API active

## Set up GPU Monitoring on a uniform Kubernetes cluster or non-Kubernetes environment

The following instructions are the basic steps to set up GPU Monitoring in the following environments:

> - In a Kubernetes cluster where **all** the nodes have GPU devices

Reviewer: ```suggestion
- In a Kubernetes cluster where **all** nodes have GPU devices
```

> 

Reviewer 2: Can we update the sentence above the Prerequisites sentence to include more context about what advanced eBPF metrics offer? Something along the lines of 

"This page provides instructions on setting up Datadog's GPU Monitoring on your infrastructure. Follow the configuration instructions that match your operating environment in the table below.

To get additional insights and advanced eBPF metrics, like GPU core utilization, you can optionally opt-in to  enabling System Probe with privileged mode."

To begin using Datadog's GPU Monitoring, your environment must meet the following criteria:

> - You have a Datadog account and are running the Datadog Agent on your infrastructure hosts. If not, see the following guides:

Reviewer 2: ```suggestion
- You are running Datadog Agent on your GPU-accelerated hosts that you want to monitor. If not, see the following guides:
```

{{< /tabs >}}

## Set up GPU Monitoring on a mixed Kubernetes cluster

> In a mixed cluster, some nodes have GPU devices while other nodes do not. Due to the way that the NVIDIA device plugin for Kubernetes works, the agent needs to have a specific runtime class to access the GPU devices that is only present on GPU nodes. Therefore, two separate DaemonSets, one for the runtime class in GPU nodes, and another for non-GPU nodes, are required.

Reviewer: ```suggestion
In a mixed cluster, some nodes have GPU devices while other nodes do not. Due to the way that the NVIDIA device plugin for Kubernetes works, the Agent needs to have a specific runtime class to access the GPU devices that is only present on GPU nodes. Therefore, two separate DaemonSets, one for the runtime class in GPU nodes, and another for non-GPU nodes, are required.
```
